### PR TITLE
change langVer to 12

### PR DESCRIPTION
**Pull request by kellyyangsong**
related to https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/2184

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -72,7 +72,7 @@
   </PropertyGroup>
 
   <PropertyGroup Condition="'$(TargetFramework)' == 'net472' Or '$(TargetFramework)' == 'net462' Or '$(TargetFramework)' == 'netstandard2.0' Or '$(TargetFramework)' == 'netcoreapp3.1'">
-    <LangVersion>11</LangVersion>
+    <LangVersion>12</LangVersion>
```
Comment (FuPingFranco) [inline Directory.Build.props:75]: Do we want to continue to update this manually moving forward or use 'preview' instead to always be up to date?

**Review by jmprieur — APPROVED** [review]

File: Directory.Build.props (hunk 1 above)
Comment (jmprieur) [reply Directory.Build.props:75]: With preview don't we risk that a preview compiler produces binaries which are not right? Would they be meant for prod?

Comment (FuPingFranco) [reply Directory.Build.props:75]: There's that risk but didn't we used "latest" before which technically had the same risk? also, it does allow us to catch any issues from a new C# version earlier and not later.

Comment (kellyyangsong) [reply Directory.Build.props:75]: "latest" did not infer the same risk and maps to pre-11

**Review by FuPingFranco — APPROVED** [review]

**Pull request merged**